### PR TITLE
Various compilation warning fixes

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: steamfriends.cpp [hunk 1]
```diff
@@ -92,7 +94,7 @@ bool SteamFriends_Invoke(void *SteamFriendsContext, echandle MethodDictionaryHan
         RetVal = IDictionary_GetInt32ByKey(MethodDictionaryHandle, "index", &Value32);
         if (RetVal == true)
             SteamFriends_GetFriendByIndex(Value32, (uint64_t *)&Value64);
-        String_Print(Value64String, Element_Count(Value64String), "%lld", Value64);
+        String_Print(Value64String, Element_Count(Value64String), "%" PRIu64, (uint64_t)Value64);
```
Comment (steve-tucker) [inline steamfriends.cpp:97]: I don't like the PRIu64 being an addendum to the string, is there anyway to put it in the string?

Comment (sergio-nsk) [reply steamfriends.cpp:97]: I like it. Also always wanted to do it. It enabled compiler warnings if wrong arguments are used.

File: steamuser.cpp [hunk 2]
```diff
@@ -17,10 +18,10 @@ class UserResults {
     CCallbackManual<UserResults, GetAuthSessionTicketResponse_t, false> AuthSessionTicket;
     void OnAuthSessionTicketResponse(GetAuthSessionTicketResponse_t *Response);
 
-    void *User;
-    HAuthTicket Ticket;
-    char TicketData[2048];
-    uint32_t TicketSize;
+    void *User = NULL;
```
Comment (sergio-nsk) [inline steamuser.cpp:21]: `nullptr` in C++.

File: steamuser.cpp [hunk 3]
```diff
@@ -196,6 +197,8 @@ bool SteamUser_Init(void) {
     SteamUserStruct *User = NULL;
 
     User = (SteamUserStruct *)malloc(sizeof(SteamUserStruct));
+    if (User == NULL)
```
Comment (sergio-nsk) [inline steamuser.cpp:200]: `if (User)` or `if (User == nullptr` in C++.

**Review by steve-tucker — APPROVED** [review]
I don't like the PRIu64 string changes, but it's technically correct. Take your approval, damn it!

File: steamuserstats.cpp [hunk 4]
```diff
@@ -277,6 +275,8 @@ bool SteamUserStats_Init(void) {
     SteamUserStatsStruct *UserStats = NULL;
 
     UserStats = (SteamUserStatsStruct *)malloc(sizeof(SteamUserStatsStruct));
+    if (UserStats == NULL)
```
Comment (sergio-nsk) [inline steamuserstats.cpp:278]: `if (User)` or `if (User == nullptr` in C++.

**Review by sergio-nsk — CHANGES_REQUESTED** [review]

File: steamfriends.cpp (hunk 1 above)
Comment (steve-tucker) [reply steamfriends.cpp:97]: Good point but still makes it harder to read.

File: steamuser.cpp (hunk 3 above)
Comment (steve-tucker) [reply steamuser.cpp:200]: malloc isn't c++ either it should be User = new SteamUserStruct(); and delete, right? Tbh, I think both changes are pretty pedantic in this context.

**Pull request merged**